pkg/aws/ec2/ec2.go

```diff
@@ -82,17 +79,55 @@ type Config struct {
 }
 
 // New creates an ec2 collector
-func New(config *Config) *Collector {
+func New(ctx context.Context, config *Config) (*Collector, error) {
 	logger := config.Logger.With("logger", "ec2")
+	computeMap := NewComputePricingMap(logger, config)
+	storageMap := NewStoragePricingMap(logger, config)
+
+	priceTicker := time.NewTicker(config.ScrapeInterval)
+	machineTicker := time.NewTicker(config.ScrapeInterval)
+
+	// Initial population so that Collect can use the maps
+	if err := computeMap.GenerateComputePricingMap(ctx); err != nil {
+		return nil, fmt.Errorf("failed initial compute pricing: %w", err)
+	}
+	if err := storageMap.GenerateStoragePricingMap(ctx); err != nil {
+		return nil, fmt.Errorf("failed initial storage pricing: %w", err)
+	}
+
+	go func(ctx context.Context) {
```

Comment:
nit: we don't need to explicitly pass `ctx` as a parameter. It is implicitly available to the goroutine from the outer scope
```suggestion
	go func() {
```